dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java

```diff
@@ -0,0 +1,119 @@
+package com.dotmarketing.image.focalpoint;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.apache.commons.io.IOUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.fileassets.business.FileAssetAPI;
+import com.dotmarketing.util.Logger;
+import io.vavr.control.Try;
+
+public class FocalPointAPIImpl implements FocalPointAPI {
+
+    final FocalPointCache cache;
+    final FileAssetAPI fileAssetAPI;
+
+    public FocalPointAPIImpl() {
+        this(APILocator.getFileAssetAPI(), new FocalPointCache());
+
+    }
+
+    public FocalPointAPIImpl(FileAssetAPI fileAssetAPI, FocalPointCache cache) {
+        this.fileAssetAPI = fileAssetAPI;
+        this.cache = cache;
+    }
+
+
+    private File getFPFile(String inode, String fieldVar) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'inode' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4761083)